### PR TITLE
788 add missing pact contract tests referrals and course participations 

### PR DESCRIPTION
**Pull request by anandelangovan1**
## Context

https://trello.com/c/gB76GqkW

## Changes in this PR

Pact tests had been disabled in the past. As part of this ticket we have updated the pact provider tests to satisfy the consumer. 

## Release checklist

[Release process documentation](../doc/how-to/perform-a-release.md)

As part of our continuous deployment strategy we must ensure that this work is
ready to be released once merged.

### Pre-merge

- [ ] There are changes required to the Accredited Programmes UI for this change to work...
  - [ ] ... and they been released to production already

### Post-merge

- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-preprod-environment) release to preprod
- [ ] [Manually approve](../doc/how-to/perform-a-release.md#releasing-to-the-production-environment) release to prod

### CODE REVIEW

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt [hunk 1]
```diff
@@ -15,8 +15,6 @@ import org.springframework.context.annotation.Import
 import org.springframework.http.HttpHeaders
 import org.springframework.test.context.ActiveProfiles
 import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.common.config.JwtAuthHelper
-import uk.gov.justice.digital.hmpps.hmppsaccreditedprogrammesapi.domain.repository.CourseRepository
```
Comment (nsmj63-dxw) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt]: One of the things we try to do as part of the atomic-commits process is squashing later changes down that aren't directly adding functionality. For instance, you could probably use `git rebase -i main` to make fixup commits and squash down [de4ab75](https://github.com/ministryofjustice/hmpps-accredited-programmes-api/pull/155/commits/de4ab75806abfc9c1383294b2a16aaa52a6f9f8b) into any of the earlier ones. There's an excellent tool out there that can do most of this heavy lifting for you called [`git absorb`](https://github.com/tummychow/git-absorb). Happy to go through it with you at some point if you like.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt [hunk 2]
```diff
@@ -15,8 +15,6 @@ import org.springframework.context.annotation.Import
...
-
```
Comment (nsmj63-dxw) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt]: For annoying things like empty lines, trailing commas, optimising imports, and so on, this repo also contains a helpful thing called `ktLintFormat`, which runs over the codebase and makes the relevant changes. You can also set it up as a pre-commit hook - instructions for it should be in the readme.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt [hunk 3]
```diff
@@ -29,69 +29,69 @@ class PactContractTest {
...
-  fun requestToDestroyCourse(){}
+  fun `Request to destroy course participation with ID 62d5c0be-edb9-47bb-89aa-7fc489e753d0`(){}
 
   @State("A course participation exists with ID 62d5c0be-edb9-47bb-89aa-7fc489e753d0")
-  fun ensureCourseParticipationExists(){}
+  fun `Ensure course participation exists with ID 62d5c0be-edb9-47bb-89aa-7fc489e753d0`(){}
 
   @State("A course participation exists with ID 2480e7e3-5f9d-4e94-900c-148374ddbd72")
-  fun ensureCourseParticipationExistsForId_2480e7e3_5f9d_4e94_900c_148374ddbd72(){}
+  fun `ensure course participation exists with ID 2480e7e3-5f9d-4e94-900c-148374ddbd7`(){}
```
Comment (nsmj63-dxw) [inline src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt]: Silly thing to mention, I know, but it's probably worth having our capitalisation be consistent here. Stick to either `ensure` or `Ensure` but don't use both! 😁

File: src/test/resources/db/migration/R__test_data.sql [hunk 4]
```diff
@@ -16,6 +16,9 @@ VALUES
 
 INSERT INTO offering(offering_id, course_id, organisation_id, contact_email, secondary_contact_email)
 VALUES
+        ('790a2dfe-7de5-4504-bb9c-83e6e53a6537', 'd3abc217-75ee-46e9-a010-368f30282367', 'BWN', 'nobody-bwn@digital.justice.gov.uk', null),
+        ('39b77a2f-7398-4d5f-b744-cdcefca12671', 'd3abc217-75ee-46e9-a010-368f30282367', 'BXI', 'nobody-bxi@digital.justice.gov.uk', null),
```
Comment (nsmj63-dxw) [inline src/test/resources/db/migration/R__test_data.sql:20]: Fixes like this are another good place to do `fixup`s or use `git absorb`.

File: src/test/resources/db/migration/R__test_data.sql [hunk 5]
```diff
@@ -24,7 +24,7 @@ VALUES
        ('7f98826a-616c-4414-a278-525fc02505a0', 'd3abc217-75ee-46e9-a010-368f30282368', 'MDI', 'nobody-mdi@digital.justice.gov.uk', 'nobody2-mdi@digital.justice.gov.uk'),
        ('fee62dde-87f5-4dfd-9a44-e80d48f64be9', 'd3abc217-75ee-46e9-a010-368f30282369', 'MDI', 'nobody-mdi@digital.justice.gov.uk', 'nobody2-mdi@digital.justice.gov.uk');
 
-INSERT INTO referral (referral_id, offering_id, prison_number, referrer_id, reason, oasys_confirmed, has_reviewed_programme_history, status)
+INSERT INTO referral (referral_id, offering_id, prison_number, referrer_id, additional_information, oasys_confirmed, has_reviewed_programme_history, status)
```
Comment (nsmj63-dxw) [inline src/test/resources/db/migration/R__test_data.sql:27]: Another good place to do a `fixup` or `git absorb`. 😄

File: src/test/resources/db/migration/R__test_data.sql [hunk 6]
```diff
@@ -26,7 +26,29 @@ VALUES ('7fffcc6a-11f8-4713-be35-cf5ff1aee517', 'd3abc217-75ee-46e9-a010-368f302
 
        ('b328ebc8-1f7b-4236-b4ac-30f50b43a92d', '1811faa6-d568-4fc4-83ce-41118b90242e', 'BWN', 'nobody-bwn@digital.justice.gov.uk', null);
 
+INSERT INTO course_participation (course_participation_id, prison_number, course_name, detail)
+VALUES ('53787979-c455-4a3f-9787-9cd2f69a9dad', 'E6NInAW', 'Grey Course', 'Textor atrox accommodo corona volva conturbo suscipit civis.'),
+       ('1811faa6-d568-4fc4-83ce-41118b90242e', 'TPPIdBJ', 'Black Course', 'Men who deny their current offence can participate in BBR');
+
+
+--
+-- val id: UUID? = null,
```
Comment (nsmj63-dxw) [inline src/test/resources/db/migration/R__test_data.sql]: Try not to leave comments lying around unless they're absolutely necessary.

**Review by nsmj63-dxw — CHANGES_REQUESTED** [review]
Functionality seems good enough to merge in, but let's condense things down a touch so that we can have our commit history look good. Happy to pair with you on this if you think it'll be valuable.

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt (hunk 1 above)
Comment (anandelangovan1) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt]: I haven't used git absorb. I am used to using `git rebase`
Happy to pair with you on using git absorb

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt (hunk 2 above)
Comment (anandelangovan1) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt]: Yeah, the build picked it up 👍 
Something for next time

File: src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt (hunk 3 above)
Comment (anandelangovan1) [reply src/test/kotlin/uk/gov/justice/digital/hmpps/hmppsaccreditedprogrammesapi/pact/PactContractTest.kt]: I think you have reviewed an outdated file.

File: src/test/resources/db/migration/R__test_data.sql (hunk 6 above)
Comment (anandelangovan1) [reply src/test/resources/db/migration/R__test_data.sql]: I think you have reviewed an outdated file.

**Comment by anandelangovan1** [comment]
> Functionality seems good enough to merge in, but let's condense things down a touch so that we can have our commit history look good. Happy to pair with you on this if you think it'll be valuable.

**Review by Gweaton — APPROVED** [review]

**Review by anandelangovan1 — COMMENTED** [review]
Changes have been addressed

**Review by nsmj63-dxw — APPROVED** [review]
We got there!

**Pull request merged**